auth-api/src/auth_api/resources/task.py

```diff
@@ -23,27 +24,52 @@
 from auth_api.exceptions import BusinessException
 from auth_api.schemas import TaskSchema
 
-
 API = Namespace('tasks', description='Endpoints for tasks management')
 TRACER = Tracer.get_instance()
 _JWT = JWTWrapper.get_instance()
 
 
-@cors_preflight('GET,POST,OPTIONS')
-@API.route('', methods=['GET', 'POST', 'OPTIONS'])
+@cors_preflight('GET,OPTIONS')
+@API.route('/<string:task_relationship_type>', methods=['GET', 'POST', 'OPTIONS'])
 class Tasks(Resource):
-    """Resource for managing tasks."""
+    """Resource for fetching tasks."""
 
     @staticmethod
     @TRACER.trace()
     @cors.crossdomain(origin='*')
     @_JWT.has_one_of_roles(
         [Role.STAFF.value])
-    def get():
+    def get(task_relationship_type):
         """Fetch tasks."""
         try:
-            tasks = TaskService.fetch_tasks()
+            tasks = TaskService.fetch_tasks(task_relationship_type=task_relationship_type)
             response, status = {'tasks': TaskSchema().dump(tasks, many=True)}, http_status.HTTP_200_OK
         except BusinessException as exception:
             response, status = {'code': exception.code, 'message': exception.message}, exception.status_code
         return response, status
+
+
+@cors_preflight('PATCH,OPTIONS')
+@API.route('/<int:task_id>/status', methods=['PATCH', 'OPTIONS'])
+class TaskStatus(Resource):
```

Comment:
May be make it a PUT on task, PATCH may become complicated later [lesson learned from org patch].